ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -343,46 +349,56 @@ public QueryMetadata getQueryExecutionPlan(final Query query) throws Exception {
               (CreateTable) statement,
               overriddenProperties,
               topicClient,
-              false),
-          tempMetaStoreForParser
+              enforceTopicExistence),
+          tempMetaStoreForParser,
+          false
       );
       ddlCommandExec.tryExecute(
           new CreateTableCommand(
               statementString,
               (CreateTable) statement,
               overriddenProperties,
               topicClient,
-              false),
-          tempMetaStore
+              enforceTopicExistence),
+          tempMetaStore,
+          false
       );
       return new Pair<>(statementString, statement);
     } else if (statement instanceof DropStream) {
       ddlCommandExec.tryExecute(new DropSourceCommand(
-          (DropStream) statement,
-          DataSource.DataSourceType.KSTREAM,
-          this
-      ), tempMetaStore);
-      ddlCommandExec.tryExecute(
-          new DropSourceCommand((DropStream) statement, DataSource.DataSourceType.KSTREAM, this),
-          tempMetaStoreForParser
-      );
+                                    (DropStream) statement,
+                                    DataSource.DataSourceType.KSTREAM,
+                                    schemaRegistryClient),
+                                tempMetaStore,
+                                false);
+      ddlCommandExec.tryExecute(new DropSourceCommand(
+                                    (DropStream) statement,
+                                    DataSource.DataSourceType.KSTREAM,
+                                    schemaRegistryClient),
+                                tempMetaStoreForParser,
+                                false);
```

Comment:
This should be true here right? This is called from the validation phase, before the query makes it onto the command topic.

Reply:
The validation for Drop statements is done in the `KsqlResource.java` class.